tests/wallet/conftest.py

```diff
@@ -2,18 +2,85 @@
+
+
+@pytest.fixture(scope="function", autouse=True)
+async def ignore_block_validation(request: pytest.FixtureRequest, monkeypatch: pytest.MonkeyPatch) -> None:
+    if "standard_block_tools" in request.keywords:
+        return None
+
+    async def validate_block_body(*args: Any, **kwargs: Any) -> Tuple[Literal[None], NPCResult]:
+        return None, args[7]
+
+    def create_wrapper(original_create: Any) -> Any:
+        async def new_create(*args: Any, **kwargs: Any) -> Any:
+            # Modify the config argument directly since it's a mutable dictionary
+            if "config" in kwargs:
+                kwargs["config"]["single_threaded"] = True
+            else:  # pragma: no cover
+                # Nowhere covers this line right now but its entirely possible
+                args[0]["single_threaded"] = True
+
+            # Call the original function with modified arguments
+            full_node = await original_create(*args, **kwargs)
+            return full_node
+
+        return new_create
+
+    monkeypatch.setattr("chia.simulator.block_tools.BlockTools", WalletBlockTools)
+    monkeypatch.setattr(FullNode, "create", create_wrapper(FullNode.create))
+    monkeypatch.setattr("chia.consensus.blockchain.validate_block_body", validate_block_body)
+    monkeypatch.setattr(
+        "chia.consensus.block_header_validation.validate_unfinished_header_block", lambda *_, **__: (uint64(1), None)
+    )
+    monkeypatch.setattr(
+        "chia.wallet.wallet_blockchain.validate_finished_header_block", lambda *_, **__: (uint64(1), None)
+    )
+    monkeypatch.setattr(
+        "chia.consensus.multiprocess_validation.validate_finished_header_block", lambda *_, **__: (uint64(1), None)
+    )
+    monkeypatch.setattr(
+        "chia.consensus.multiprocess_validation.verify_and_get_quality_string", lambda *_, **__: bytes32([0] * 32)
+    )
+    monkeypatch.setattr("chia.consensus.block_record.BlockRecord.sp_total_iters", lambda *_: uint128(0))
+    monkeypatch.setattr("chia.consensus.block_record.BlockRecord.ip_sub_slot_total_iters", lambda *_: uint128(0))
+    monkeypatch.setattr("chia.consensus.make_sub_epoch_summary.calculate_sp_iters", lambda *_: uint64(0))
+    monkeypatch.setattr("chia.consensus.make_sub_epoch_summary.calculate_ip_iters", lambda *_: uint64(0))
+    monkeypatch.setattr("chia.consensus.difficulty_adjustment._get_next_sub_slot_iters", lambda *_: uint64(1))
+    monkeypatch.setattr("chia.consensus.difficulty_adjustment._get_next_difficulty", lambda *_: uint64(1))
+    monkeypatch.setattr("chia.full_node.full_node_store.calculate_sp_interval_iters", lambda *_: uint64(1))
+    monkeypatch.setattr("chia.consensus.pot_iterations.calculate_sp_interval_iters", lambda *_: uint64(1))
+    monkeypatch.setattr("chia.consensus.pot_iterations.calculate_ip_iters", lambda *_: uint64(1))
+    monkeypatch.setattr("chia.consensus.block_record.BlockRecord.sp_sub_slot_total_iters", lambda *_: uint64(1))
```

Comment:
is this a better approach than to make it possible to create a `BlockTool` object with flags (or perhaps just a `bool`) specifying what it needs to support and not. e.g. you could imagine creating a `BlockTools` in wallet-mode.

Reply:
That's what the patching of `BlockTools` with `WalletBlockTools` is for.  I didn't see any need to get granular with the stuff being mocked because I can't envision a scenario where the wallet cares about the stuff being mocked like VDF iterations etc.  The rest of this patching is to make the production full node code handle the mocked blocks.  Seemed like a simpler & much better alternative than piping a `test: bool = False` argument through all of our production code.

Reply:
I see, this isn't done to affect the `BlockTools` behavior, it's to make the full node accept the invalid blocks.
Having a config option `disable_block_validation` might be reasonable though.

Reply:
You mean in the full node?  That seems a bit sketchy to me but at the very least its out of the scope of the PR.

Reply:
@Quexington Can this conversation be resolved now?